nbtc/relayer.go

```diff
@@ -20,10 +21,17 @@ type Relayer struct {
 	btcClient        bitcoin.Client
 	shutdownChan     chan struct{}
 	processTxsTicker *time.Ticker
+	confirmTxsTicker *time.Ticker
 }
 
 // NewRelayer creates a new Relayer instance with the given configuration.
-func NewRelayer(btcClientConfig rpcclient.ConnConfig, processTxsInterval time.Duration, db *dal.DB) (*Relayer, error) {
+func NewRelayer(
+	btcClientConfig rpcclient.ConnConfig,
+	processTxsInterval time.Duration,
+	confirmTxsInterval time.Duration,
+	db *dal.DB,
+) (*Relayer, error) {
+
 	if db == nil {
 		return nil, fmt.Errorf("database cannot be nil")
```

Comment:
I think we should use logger more than `fmt`.

Reply:
so should i add logger? We dont want to just log the problem and conitune because the relayer without the db has no point

Reply:
I added the logging of the error but still return the error here